src/go/k8s/pkg/resources/certmanager/pandaproxy_api.go

```diff
@@ -40,6 +40,12 @@ func (r *PkiReconciler) preparePandaproxyAPI(
 ) []resources.Resource {
 	toApply := []resources.Resource{}
 
+	kn := NewCommonName(r.pandaCluster.Name, keystore)
+	keystoreKey := types.NamespacedName{Name: string(kn), Namespace: r.pandaCluster.Namespace}
+	keystoreSecret := NewKeystoreResource(r.Client, r.scheme, r.pandaCluster, keystoreKey, r.logger)
```

Comment:
can't we use the same password for all keystores? Would not that be enough maybe? 🤔

Reply:
The name of the secret and the password will be the same as the name of the secret derives from pandaCluster.Name + keystore suffix. Our resource creates only one secret. Any other execution of the `Ensure` will end up as no-op.

Reply:
ah the secret is the same? Then that's kind of not obvious from the implementation and pretty error prone for the future because the same code is copy-pasted across files. I guess then the secret should be injected in there from higher level to ensure there's only one and not re-created with (hopefully) the same name in every file

Reply:
I don't agree with the statement that passing reference is less error prone that having idempotent resource closer to the usage. 

> not re-created with (hopefully) the same name in every file

I agree that the name of the `keystoreKey` could be error prone.

I move the whole creation to the pki (one level up). Please take a look.